mkosi/config.py

```diff
@@ -795,6 +794,61 @@ def cache_manifest(self) -> dict[str, Any]:
         return manifest
 
 
+def parse_ini(path: Path, only_sections: Sequence[str] = ()) -> Iterator[tuple[str, str, str]]:
+    section: Optional[str] = None
+    setting: Optional[str] = None
+    value: Optional[str] = None
+
+    for l in textwrap.dedent(path.read_text()).splitlines():
+        for c in ("#", ";"):
+            comment = l.find(c)
+            if comment >= 0:
+                l = l[:comment]
+
+        if not l.strip():
+            continue
+
+        # If we have a section, setting and value, any line that's indented is considered part of the
+        # setting's value.
+        if section and setting and value is not None and l[0].isspace():
+            value = f"{value}\n{l.strip()}"
+            continue
+
+        # So the line is not indented, that means we either found a new section or a new setting. Either way,
+        # let's yield the previous setting and its value before parsing the new section/setting.
+        if section and setting and value is not None:
+            yield section, setting, value
+            setting = value = None
+
+        l = l.strip()
+
+        if l[0] == '[':
+            if l[-1] != ']':
+                die(f"{l} is not a valid section")
+
+            section = l[1:-1]
```

Comment:
```suggestion
            section = l[1:-1].strip()
```
otherwise all whitespace (but non-empty) section names are possible. With the strip the below error message would become wrong, though.